### PR TITLE
Add to planning action refactoring

### PR DESCRIPTION
**Pull request by thecalcc**
SDESK-6267

### CODE REVIEW

**Comment by MarkLark86** [comment]
@thecalcc What's the ticket for this one?

**Comment by thecalcc** [comment]
Not sure if it's right to use `EditCoverageAssignmentModalComponent` for `coverage_assignment_status`. Let me know if this is the proper component to be used.

**Comment by MarkLark86** [comment]
> coverage_assignment_status

Hey @thecalcc It looks like the `EditCoverageAssignmentModalComponent` is not used anywhere in the current code (in `develop`). So it can be removed.

I'm not sure what you mean by usage of `coverage_assignment_status`. All item action modals are used from [ItemActionConfirmation/modal.tsx](https://github.com/superdesk/superdesk-planning/blob/develop/client/components/ItemActionConfirmation/modal.tsx) file

File: client/controllers/AddToPlanningController.tsx [hunk 1]
```diff
@@ -219,15 +224,19 @@ export class AddToPlanningController {
                         this.notify.error(err);
                     });
 
-                    this.$scope.resolve('foo');
+                    if (!authoringReactViewEnabled) {
+                        this.$scope.resolve('foo');
```
Comment (tomaskikutis) [inline client/controllers/AddToPlanningController.tsx:228]: shouldn't you return after this and similar occurences in this file?

File: client/planning-extension/src/extension.ts [hunk 2]
```diff
@@ -13,6 +13,9 @@ import {AssignmentsList} from './assignments-overview';
 import {IPlanningExtensionConfigurationOptions} from './extension_configuration_options';
 import {AutopostIngestRuleEditor} from './ingest_rule_autopost/AutopostIngestRuleEditor';
 import {AutopostIngestRulePreview} from './ingest_rule_autopost/AutopostIngestRulePreview';
+import ng from 'superdesk-core/scripts/core/services/ng';
+// import {superdeskApi} from '../../../client/superdeskApi';
+// import {isContentLinkToCoverageAllowed} from '../../../client/utils/archive';
```
Comment (tomaskikutis) [inline client/planning-extension/src/extension.ts:18]: remove comments

File: client/planning-extension/src/extension.ts [hunk 3]
```diff
@@ -112,6 +115,34 @@ const extension: IExtension = {
                 contributions: {
                     entities: {
                         article: {
+                            getActions: (item: IArticle) => [{
+                                label: 'Add to Planning',
```
Comment (tomaskikutis) [inline client/planning-extension/src/extension.ts:119]: use gettext

File: client/planning-extension/src/extension.ts [hunk 4]
```diff
@@ -112,6 +115,34 @@ const extension: IExtension = {
...
+                                    const privilegesService = ng.get('privileges');
+                                    const archiveService = ng.get('archiveService');
+                                    const authoringService = ng.get('authoring');
+
+                                    if (
+                                        privilegesService.userHasPrivileges({planning_planning_management: 1}) &&
+                                        privilegesService.userHasPrivileges({archive: 1}) &&
+                                        !item.assignment_id != null &&
+                                        !archiveService.isPersonal(item) &&
+                                        // !superdeskApi.entities.article.isLockedInOtherSession(item) &&
```
Comment (tomaskikutis) [inline client/planning-extension/src/extension.ts:132]: if it has to be here - uncomment - otherwise, remove

**Comment by tomaskikutis** [comment]
I guess this might have got autoclosed due to deleting the branch. I assume it is still relevant @thecalcc ?

**Comment by thecalcc** [comment]
> I guess this might have got autoclosed due to deleting the branch. I assume it is still relevant @thecalcc ?
Yes, still relevant

**Comment by tomaskikutis** [comment]
Github doesn't allow to reopen it since target branch was deleted. Can you open a fresh PR against develop and link to this one?

**Pull request closed without merging**